VoodooI2CHID/VoodooI2CPrecisionTouchpadHIDEventDriver.hpp

```diff
@@ -23,6 +24,14 @@
 #define INPUT_MODE_MOUSE 0x00
 #define INPUT_MODE_TOUCHPAD 0x03
 
+#define CATALINA_MAJOR_VERSION 19 // Darwin major version for Catalina
+
+typedef struct __attribute__((__packed__)) {
+     UInt8 reportID;
```

Comment:
A question - `setReport` appears to skip first byte if it equals to the report ID. Why adding it to the buffer, then?

By the HID spec it should be there and I2C-HID just ignores it as it's duplicate?

Reply:
On some devices, such as @leebinder's, the report ID for input mode is 3. This is same as the trackpad mode, so the mode byte gets skipped.

Reply:
We can probably trackdown the issue in a follow up PR. Right now, as it stands, older versions of macOS do not adhere to the HID spec and this is a decent enough work around.

Reply:
Do you know if calling `input_mode->setValue()` adds the report ID first, so that issue wouldn't happen when we can fully transition to it?

Reply:
I'm fairly certain it does. Looking at AlpsHID as well, every report it sends using setReport has the reportID in the first byte as well. This probably means that the USB HID stack assumes that the report ID is in the first byte. Not sure what breaks down in Catalina though.